httpclient5/src/test/java/org/apache/hc/client5/http/impl/cookie/TestPublicSuffixListParser.java

```diff
@@ -111,7 +111,7 @@ void testParseLocal() {
 
         cookie.setDomain(".blah");
         cookie.setAttribute(Cookie.DOMAIN_ATTR, ".blah");
-        Assertions.assertTrue(filter.match(cookie, new CookieOrigin("somehost.blah", 80, "/stuff", false)));
+        Assertions.assertTrue(filter.match(cookie, new CookieOrigin("somehost.blah", 80, "/stuff", false), true));
```

Comment:
We shouldn't need to use legacyMode here, as the test PSL file doesn't contain any `DomainType.PRIVATE` entries (or `DomainType.ICANN` entries for that matter - it seems to be parsed with a separate parser that doesn't handle the sections and sets everything to `DomainType.UNKNOWN`?).

Has the behavior changed because of our proper implementation of the default wildcard rule perhaps?

Honestly, I can't tell what the cookie `PublicSuffixDomainFilter` is supposed to be doing, or why some of these tests (presumably) work... :/

Reply:
We can go through this together with the debugger if you'd like but we hit this return statement in the matcher if we don't go for legacy mode in this case:
```
                if (nextSegment == null && (expectedType == null || expectedType == DomainType.UNKNOWN)) {
                    return result;
                }
```

Reply:
Sorry, my second comment was sitting there in "pending" state - I thought I'd sent it...

tl;dr: When the input cookie domain is `.blah`  I think this code (in non-legacy mode) will return `null`, which will be interpreted as `blah` being a public suffix (which under the default `*` rule, it is), and the cookie filter will return false. IMO that's correct, and the test is wrong - it should assert `false`.

Reply:
OK, so let me revert the usage of `legacyMode` for this thing and change this test to `AssertFalse` for now.